### PR TITLE
click until text step

### PR DESCRIPTION
**Pull request by ALegchilov**
## Description

Please describe your changes and any new features you're introducing, or issues you're fixing.

- [x] Updated the CHANGELOG.md.
- [x] Updated package version in package.json, package-lock.json.
- [x] Checked that there aren't other open pull requests for the same issue/update.
- [x] Checked that your contribution follows the project's contribution guidelines.
- [x] Added corresponding unit/E2E tests
- [x] Unit and E2E pass

### Related Issues

[Please link any related issues or bug reports that this PR will address.](https://github.com/qavajs/steps-playwright/issues/125)

### CODE REVIEW

**Review by AlexGalichenko — CHANGES_REQUESTED** [review]

File: src/IQavajsPlaywrightConfig.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    /**
+     * Element presence timeout
+     *
+     * default: 10_000
+     * @example
+     * export default {
+     *     present: 20_000
+     * }
+     */
+    present?: number,
```
[the remark this answers is not in the document]
Comment (ALegchilov) [reply src/IQavajsPlaywrightConfig.ts]: Coud you check now?

File: src/waits.ts [hunk 2]
```diff
@@ -82,18 +83,90 @@ When(
...
+      await elementToClick.waitFor({state: 'attached', timeout});
+      const expectedText = await getValue(value);
+      const poll = getPollValidation(validationType);
+      await poll(
+        async () => {
+          await elementToClick.click();
+          return elementToCheck.innerText();
+        },
+        expectedText,
+        {timeout, interval: defaultTimeouts.actionInterval},
```
Comment (AlexGalichenko) [inline src/waits.ts:131]: here default timeout should be imported from config

File: utils.d.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { Locator } from '@playwright/test';
 export function getElement(alias: string): Promise<Locator>;
 export function getValue(alias: string): any;
-export function getConditionWait(condition: string): (element: Locator) => Promise;
+export function getConditionWait(condition: string): (element: Locator) => Promise<Function>;
```
Comment (AlexGalichenko) [inline utils.d.ts:4]: Promise<void>

File: src/waits.ts [hunk 4]
```diff
@@ -82,18 +83,90 @@ When(
...
+      await elementToClick.waitFor({state: 'attached', timeout});
+      const expectedValue = await getValue(value);
+      const poll = getPollValidation(validationType);
+      await poll(
+        async () => {
+            await elementToClick.click();
+            return elementToCheck.inputValue();
+            },
+            expectedValue,
+            {timeout, interval: defaultTimeouts.actionInterval},
```
Comment (AlexGalichenko) [inline src/waits.ts:167]: here default timeout should be imported from config

**Review by AlexGalichenko — CHANGES_REQUESTED** [review]

**Review by AlexGalichenko — APPROVED** [review]

File: src/IQavajsPlaywrightConfig.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      /**
+       * Interval for steps refreshing page in loop unless some validation to be met
+       *
+       * default: 2_000
+       * @example
+       * export default {
+       *     pageRefreshInterval: 10_000
+       * }
+       */
+      pageRefreshInterval: number
```
Comment (KirillBogdanets) [inline src/IQavajsPlaywrightConfig.ts:77]: why is it the only mandatory property?

Comment (ALegchilov) [reply src/IQavajsPlaywrightConfig.ts:77]: Good catch. It's optional indeed.

File: utils.d.ts (hunk 3 above)
Comment (ALegchilov) [reply utils.d.ts]: Agreed to declare it as Promise<void>

**Pull request merged**